core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -164,7 +164,7 @@ public int run(final String[] args,
             e.printStackTrace(System.err);
         } finally {
             if (kafkaAdminClient != null) {
-                kafkaAdminClient.close(60, TimeUnit.SECONDS);
+                kafkaAdminClient.close(java.time.Duration.ofSeconds(60));
```

Comment:
We cannot import `java.time.Duration` because we already import `javax.xml.datatype.Duration`.

I was wondering if we could get rid of `javax.xml.datatype.Duration` though and replace it with `java.time.Duration` that was not available in Java7 when we added `javax.xml.datatype.Duration`.

Thoughts? \cc @jeqo who extended the reset tool via #3831

Reply:
The only place we need this duration is

```
duration.negate().addTo(now);
```

`java.time.Duration` has `negated` but its `addTo` is taking a `Temporal`, and all its extends like `LocalTime` etc do not have the right API of `getTime`. So I think we just keep it like this.

Reply:
We should remove the javax.xml.datatype.Duration usage since it's not part of the Java base module.

Reply:
I was locking into the corresponding KIPs
 - https://cwiki.apache.org/confluence/display/KAFKA/KIP-122%3A+Add+Reset+Consumer+Group+Offsets+tooling
 - https://cwiki.apache.org/confluence/display/KAFKA/KIP-171+-+Extend+Consumer+Group+Reset+Offset+for+Stream+Application

Both say:
```
--by-duration  PnDTnHnMnS

Duration must be specified in ISO8601 format.
```

Looking to JavaDocs of `java.time.Duration#parse()` it says:
```
     * Obtains a {@code Duration} from a text string such as {@code PnDTnHnMn.nS}.
     * <p>
     * This will parse a textual representation of a duration, including the
     * string produced by {@code toString()}. The formats accepted are based
     * on the ISO-8601 duration format {@code PnDTnHnMn.nS} with days
     * considered to be exactly 24 hours.
```

Thus, it seems save to replace
```
final javax.xml.datatype.Duration durationParsed = DatatypeFactory.newInstance().newDuration(duration);
```
with
```
final java.time.Duration durationParsed = java.time.Duration.parse(duration);
```

Btw: we should do the same for `kafka.admin.ConsumerGroupCommand` (I can add this change to this PR).

Thoughts?

Reply:
@mjsax , I agree, it seems safe and desirable to swap them out.